crates/compiler/gen_llvm/src/llvm/bitcode.rs

```diff
@@ -54,7 +67,35 @@ fn call_bitcode_fn_help<'ctx>(
     args: &[BasicValueEnum<'ctx>],
     fn_name: &str,
 ) -> CallSiteValue<'ctx> {
-    let it = args.iter().map(|x| (*x).into());
+    let it = args
+        .iter()
+        .map(|x| {
+            if env.target_info.operating_system == roc_target::OperatingSystem::Windows {
+                if x.get_type() == env.context.i128_type().into() {
+                    let parent = env
+                        .builder
+                        .get_insert_block()
+                        .and_then(|b| b.get_parent())
+                        .unwrap();
+
+                    let alloca = create_entry_block_alloca(
+                        env,
+                        parent,
+                        x.get_type(),
+                        "pass_u128_by_reference",
+                    );
+
+                    env.builder.build_store(alloca, *x).unwrap();
+
+                    alloca.into()
+                } else {
```

Comment:
Nit: can we avoid the duplicate else with the same return value? Just merge the two ifs?